packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page.tsx

```diff
@@ -1,49 +1,55 @@
 import { useRouter } from "next/router";
 import { useEffect, useState } from "react";
+import {
+  EntityWithMetadata,
+  Subgraph,
+  SubgraphRootTypes,
+} from "@hashintel/hash-subgraph";
 import { useBlockProtocolGetEntity } from "../../../components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolGetEntity";
 import { useLoggedInUser } from "../../../components/hooks/useAuthenticatedUser";
 import { getPlainLayout, NextPageWithLayout } from "../../../shared/layout";
 import { EntityEditor } from "./[entity-id].page/entity-editor";
 import { EntityPageLoadingState } from "./[entity-id].page/entity-page-loading-state";
 import { EntityPageWrapper } from "./[entity-id].page/entity-page-wrapper";
-import {
-  extractEntityRoot,
-  RootEntityAndSubgraph,
-} from "../../../lib/subgraph";
 import { PageErrorState } from "../../../components/page-error-state";
+/** @todo - This should be moved somewhere shared */
+import { useRouteNamespace } from "../types/entity-type/use-route-namespace";
 
 const Page: NextPageWithLayout = () => {
   const router = useRouter();
+  const namespace = useRouteNamespace();
   const { authenticatedUser } = useLoggedInUser();
   const { getEntity } = useBlockProtocolGetEntity();
 
-  const [rootEntityAndSubgraph, setRootEntityAndSubgraph] =
-    useState<RootEntityAndSubgraph>();
+  const [entitySubgraph, setEntitySubgraph] =
+    useState<Subgraph<SubgraphRootTypes["entity"]>>();
   const [loading, setLoading] = useState(true);
 
   useEffect(() => {
-    const init = async () => {
-      try {
-        const entityId = router.query["entity-id"] as string;
+    if (namespace) {
+      const init = async () => {
+        try {
+          const entityUuid = router.query["entity-id"] as string;
 
-        const { data: subgraph } = await getEntity({ data: { entityId } });
+          const { data: subgraph } = await getEntity({
+            data: `${namespace.accountId}%${entityUuid}`,
```

Comment:
Should we have some way to construct these IDs in a shared place? Would make swapping out the implementation easier

Reply:
I thought about this too, I've made it :)